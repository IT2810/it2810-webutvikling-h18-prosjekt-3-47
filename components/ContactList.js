import React from 'react';
import {
    Image,
    Platform,
    ScrollView,
    StyleSheet,
    Text,
    TouchableOpacity,
    View,
    ListView,
    TouchableHighlight,
    AsyncStorage,
} from 'react-native';

export default class ContactList extends React.Component {

    constructor(props) {
        super(props);
        this.ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
    }

    render() {
        return (
            <ListView
            dataSource={
                this.ds.cloneWithRows([
                    {name: "JÃ¸rgen", tlf: "12345678", id: 1},
                    {name: "Sigurd", tlf: "87654321", id: 2}
                ])
            }
            renderRow={
                (rowData) => {
                    return (
                        <TouchableHighlight onPress={() => console.log(rowData)}>
                            <Text>{rowData.name}</Text>
                        </TouchableHighlight>
                    )
                }
            }/>
        )
    }
}
