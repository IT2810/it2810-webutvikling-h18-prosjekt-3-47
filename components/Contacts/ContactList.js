/**
 * ContactList displays a list of the users contacts. The contacts are sent as props by the ContactsScreen component
 * on rendering, and are displayed using a ListView component.
 * By selecting one contact from the list the a popup with the contacts phone number and address is displayed.
 */
import React from 'react';
import {StyleSheet, Text, ListView, View, TouchableOpacity,} from 'react-native';

import PopupDialog from "react-native-popup-dialog";
import DialogTitle from "react-native-popup-dialog/src/components/DialogTitle";

import DisplayContact from "./DisplayContact";

export default class ContactList extends React.Component {

    constructor(props) {
        super(props);
        this.ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
        this.state = {
            contact: {
                name: null,
                phoneNumber: null,
                address: null
            }
        };
    }

    openContact(contact) {
        this.setState(() => ({
            contact: contact
        }));
        this.popupDialog.show()
    };

    render(dataBlob, rowIdentities) {
        return (
            <View style={styles.container}>
                {/* Setting `enableEmptySections` to surpress warning which will be gone in a future release
                https://facebook.github.io/react-native/docs/listview.html#enableemptysections                */}
                <ListView enableEmptySections={true}
                    dataSource = {this.ds.cloneWithRows(this.props.contacts, rowIdentities)}
                    renderRow={
                        (rowData) => {
                            return (
                                <View>
                                    <TouchableOpacity
                                        onPress={() => {
                                            this.openContact(rowData)}}>
                                        <Text style={styles.text}>{rowData.name}</Text>
                                    </TouchableOpacity>
                                </View>
                            )
                        }
                    }
                    renderSeparator={(sectionId, rowIdentities) => <View key={rowIdentities} style={styles.separator} />}
                />
                <PopupDialog
                    dialogTitle={<DialogTitle title={this.state.contact.name} />}
                    ref={(popupDialog) => {
                        this.popupDialog = popupDialog;
                    }}>
                    <DisplayContact contact={this.state.contact} />
                </PopupDialog>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: '3%',
        flexDirection: 'row',
        alignItems: 'center',
    },
    text: {
        marginLeft: '3%',
        fontSize: 18,
        marginTop: '3%',
        marginBottom: '3%',
    },
    separator: {
        flex: 1,
        height: StyleSheet.hairlineWidth,
        backgroundColor: '#8E8E8E',
    },
});
