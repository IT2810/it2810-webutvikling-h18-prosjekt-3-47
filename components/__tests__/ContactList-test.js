import 'react-native';
import React from 'react';
import ContactList, {mockOpenContact} from '..Contacts/ContactList';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import * as jest from "react-native";

jest.mock('../ContactList');

let shallowRenderer = new ShallowRenderer();

let contacts = [
    {
        name: "Navn Navnesen 1",
        phoneNumber: 12345678,
        address: "Adresse 1"
    },
    {
        name: "Navn Navnesen 2",
        phoneNumber: 87654321,
        address: "Adresse 2"
    }
];


beforeEach(() => {
    // Clear all instances and calls to constructor and all methods:
    ContactList.mockClear();
    mockOpenContact.mockClear();
});

it('renders correctly', () => {
    const tree = renderer.create(<ContactList contacts={this.contacts} />).toJSON();

    expect(tree).toMatchSnapshot();
});

it('renders shallow correctly', () => {
    shallowRenderer.render(<ContactList contacts={this.contacts} />);
    const result = shallowRenderer.getRenderOutput();

    expect(result).toMatchSnapshot();
});

test('openContact', () => {
    let contactList = new ContactList(contacts[0]);
    expect(contactList.state.contact).toBeNull();

    let contact = contacts[1];
    contactList.openContact(contact);
    expect(contactList.state.contact).toEqual(contact);
});