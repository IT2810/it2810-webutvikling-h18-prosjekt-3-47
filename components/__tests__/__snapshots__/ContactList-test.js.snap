// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "padding": "3%",
    }
  }
/>
`;

exports[`Rendering renders shallow correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "padding": "3%",
    }
  }
>
  <ListViewMock
    enableEmptySections={true}
    renderRow={[Function]}
    renderScrollComponent={[MockFunction]}
    renderSeparator={[Function]}
  />
  <PopupDialog
    dialogTitle={
      <DialogTitle
        hasTitleBar={true}
        title={null}
        titleAlign="center"
      />
    }
  >
    <DisplayContact
      contact={
        Object {
          "address": null,
          "name": null,
          "phoneNumber": null,
        }
      }
    />
  </PopupDialog>
</Component>
`;
